src/data_loader.py

```
@@ -3,39 +3,388 @@
+                    values.append(value)
+                
+                # Include city_id if present
+                if city_id is not None:
+                    columns.append("city_id")
+                    values.append(city_id)
+
+                # Step 3: Build INSERT query
+                cols_str = ",".join(columns)
+                placeholders = ",".join(["%s"] * len(values))
+
+                query = f"INSERT INTO {table_name} ({cols_str}) VALUES ({placeholders})"
```

Ruff (0.13.3)

Possible SQL injection vector through string-based query construction

(S608)